source/Calamari.Common/Features/StructuredVariables/YamlEventStreamClassifier.cs

```diff
@@ -126,6 +126,19 @@ public static Scalar ReplaceValue(this Scalar scalar, string newValue)
                              scalar.Start,
                              scalar.End);
         }
+
+        public static Comment RestoreLeadingSpaces(this Comment comment)
+        {
+            var leadingSpaces = comment.Start.Line == comment.End.Line
+                ? comment.End.Column - comment.Value.Length - comment.Start.Column - 2
```

Comment:
Would be nice to have a comment here about the emitter limitation causing a space after the `#`